### PR TITLE
TCP stealing with proxy

### PR DESCRIPTION
**Pull request by DmitryDodzin**
relevant #397

### CODE REVIEW

File: mirrord-agent/src/steal.rs [hunk 1]
```diff
@@ -106,6 +105,47 @@ impl Drop for SafeIpTables {
...
+
+    fn redirect_rule(&self, redirected_port: Port, target_port: Port) -> String {
+        let redirect_rule = format!(
+            "-m tcp -p tcp --dport {} -j REDIRECT --to-ports {}",
+            redirected_port, target_port
+        );
+
+        match self {
+            IPTableFormatter::Normal => redirect_rule,
+            IPTableFormatter::Linkerd => format!("-o lo {}", redirect_rule),
```
Comment (aviramha) [inline mirrord-agent/src/steal.rs:144]: I think somewhat better implementation would be a trait then func that returns the appropriate struct (by using `dyn Impl`?) but it's a bit a nitpick ofc.
also some UT here would be nice

**Review by aviramha — APPROVED** [review]

File: mirrord-agent/src/steal.rs (hunk 1 above)
Comment (aviramha) [reply mirrord-agent/src/steal.rs:144]: you can make a trait and accept stuff that `impl` it. maybe @meowjesty have better idea

File: mirrord-agent/Cargo.toml [hunk 2]
```diff
@@ -42,6 +42,7 @@ faccess = "0.2"
 bytes = "1.2"
 regex = "1"
 socket2 = "0.4"
+mockall = "0.11"
```
Comment (aviramha) [inline mirrord-agent/Cargo.toml:45]: add this as a dev dependency

**Review by aviramha — APPROVED** [review]
LGTM! Nitpick on the cargo.toml.

File: mirrord-agent/Cargo.toml (hunk 2 above)
Comment (DmitryDodzin) [reply mirrord-agent/Cargo.toml:45]: I can't because the `mockall::automock` macro exists on the IPTables struct

Comment (DmitryDodzin) [reply mirrord-agent/Cargo.toml:45]: NVM found how to do it

**Pull request merged**